docs/developer-guide/syntax-reference.rst

```diff
@@ -788,6 +788,77 @@ The WITH clause supports the following properties:
     :start-after: Avro_note_start
     :end-before: Avro_note_end
 
+.. _create-type:
+
+CREATE TYPE
+-----------
+
+**Synopsis**
+
+.. code:: sql
+
+    CREATE TYPE <type_name> AS <type>;
+
+**Description**
+
+Create an alias for a complex type declaration.
+
+The CREATE TYPE statement registers a type alias directly in KSQL. Any types
+registered by using this command can be leveraged in future statements. The
+CREATE TYPE statement works in interactive and headless modes.
+
+Any attempt to register the same type twice, without a corresponding DROP TYPE
+statement, will fail.
+
+**Example**
+
+.. code:: sql
+
+    CREATE TYPE ADDRESS AS STRUCT<number INTEGER, street VARCHAR, city VARCHAR>;
```

Comment:
Should we follow up this example with an example usage of the custom type, in order to drive home the point? Not sure if the syntax reference is the best place for that though.

Reply:
+1 the example I've been using is to then define:
```
CREATE TYPE PERSON AS STRUCT<name VARCHAR, address ADDRESS>;
```